source/getting-started/control-with-amdc/encoder-fb/index.md

```diff
@@ -2,85 +2,96 @@
 
+To ensure that the obtained encoder offset is correct, the user may perform additional validation. For a permanent magnet synchronous motor, this can be done as follows:
 
-    // Wrapping to ensure within bounds
-    while (theta_m_enc > PI2) {
-        theta_m_enc -= PI2;
-    }
-    return theta_m_enc;
-}
-```
+1. Spin the motor up-to a steady speed under no load conditions
+1. Measure the d-axis voltage commanded by the current regulator
+1. Repeat the experiment for a few different motor speeds
+1. Plot the d-axis voltage against the motor speed
+1. The d-axis voltage if the offset is tuned correctly, should be close to zero for all speeds.
+1. In-case there is an error in the offset value, a significant speed dependent voltage will appear on the d-axis voltage
 
-### Finding the offset
 
-The example code shown above makes uses of an encoder offset value. It is necessary for the user to find this offset experimentally for their machine. For synchronous machines, this offset is the count value measured by the encoder when the rotor d-axis is aligned with the phase U of the stator. 
 
-To get the rotor to align with the phase U of stator the user may have some current flow through phase U and out of phase V and phase W. Alternately, the user may also inject some Id current with the AMDC injection function but with the rotot position set to 0.
+## Computing Speed from Position
 
-- Converting from raw counts to "theta"
-- Direction: +/- depending on phase connections
-- Sync machines: dq offset
+The user needs to compute a rotor speed signal from the obtained position signal to be used in the control algorithm. There are several ways to do this. A simple and straightforward way to do this would be to compute the discrete time derivative of the position signal in the controller as shown below. This can be referred to as $\Omega_{raw}$.
 
-## Computing Speed from Position
+$$
+\Omega_\text{raw}[k] = \frac{\theta_m[k] - \theta_m[k-1]}{T_s} 
+$$
+
+
+$\Omega_\text{raw}$ will be a choppy and noisy signal due to the derivative operation. A low pass filter may be applied to this signal as shown below to get a filtered speed, $\Omega_\text{lpf}$. The user may select an appropriate bandwidth, $\omega_b$ for the low pass filter to eliminate the noise introduced by the derivative operation. However, this signal will always be a lagging estimate of the actual rotor speed due to the characterstics of a low pass filter. 
```

Comment:
"due to the derivative operation and the digital nature of the incremental encoder"

Reply:
Can you provide some guidance on selecting the filter bandwidth? What would be a good first choice for someone to use?